### ISSUE TITLE
Unable to use passed context in graphQL query

### ISSUE THREAD

**Issue by beahuang** [pixelstew/gatsby-paginate#29]
## Expected Behavior
I pass in a string id context:

```
 createPaginatedPages({
          edges: node.thought,
          createPage: createPage,
          pageTemplate: tagTemplate,
          context: {
            id: node.id,
          },
          pageLength: 2,
          pathPrefix: `tags/${node.slug}`,
          buildPath: (index, pathPrefix) =>
            index > 1 ? `${pathPrefix}/${index}` : `/${pathPrefix}`,
        });
```

whose template has the following query:
``` 
export const pageQuery = graphql`
  query tagQuery($id: String!) {
    contentfulTag(id: { eq: $id }) {
      … fields
    }
  }
`;
```
but I'm getting a GraphQL error saying `GraphQLError: Variable "$id" of required type "String!" was not provided.` most likely because its not seeing the id variable because its inside of `additionalContext`

## Possible Solution
I’m able to pass $id in and use it in my query when i assign context differently:
```
 context: Object.assign(
        {
          group,
          pathPrefix,
          first: isFirstPage(index),
          last: isLastPage(index, groups),
          index: index + 1,
          pageCount: groups.length,
        },
        context
      ),
```
Is there a way to rewrite my graphQL query so that i can use the additionalContext?

**Comment by gregseed** [comment]
Bump!

**Comment by weedrater** [comment]
Bump!

**Comment by antoniofull** [comment]
I am stuck with this problem too. 
Is there a way to pass a variable to the context so that we can reuse it in the query? Thanks!
@beahuang where exactly are you assigning the context? It won't work in the gatsby-node file like this as context is not defined

**Comment by janosh** [comment]
I published `gatsby-paginate` [v1.1.0-beta.0](https://www.npmjs.com/package/gatsby-paginate/v/1.1.0-beta.0) to resolve this issue. Could you try `npm install gatsby-paginate@beta` and let me know if you experience any issues?

**Comment by antoniofull** [comment]
@janosh it is working, thank you!

**Comment by janosh** [comment]
@antoniofull Thanks for letting me know. This is now published as [v1.1.0](https://www.npmjs.com/package/gatsby-paginate).
